security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -58,8 +58,8 @@ type citadelClient struct {
 }
 
 // NewCitadelClient create a CA client for Citadel.
-func NewCitadelClient(endpoint string, tls bool, rootCert []byte, clusterID string) (security.Client, error) {
-	c := &citadelClient{
+func NewCitadelClient(endpoint string, tls bool, rootCert []byte, clusterID string) (*CitadelClient, error) {
```

Comment:
What is the reason of exposing citadelClient and return CitadelClient instead of the abstract interface?

Reply:
"Accept Interfaces, Return Struct" is a common pattern in golang. See  https://mycodesmells.com/post/accept-interfaces-return-struct-in-go for example

The benefit is users that want to deal with CitadelClient client can use it as such, users who want to use it only as a `security.Client` can also use it as such. This makes testing easier as well, and decouples the code from the interface